setting up merged route /forecast/{gsp_id}/{only_values}

rachel-labri-tipton opened the pull request:
# Pull Request

## Description

I've worked on adding a route to the gsp.py file that merges the routes get forecast for one gsp and get latest gsp. 

Fixes #131 task of merging the two routes. 

Image of updated routes: 

<img width="1467" alt="Screenshot 2022-09-15 at 13 21 57" src="https://user-images.githubusercontent.com/86949265/190396565-6b73964c-e859-46f1-a560-29ac64819bff.png">



Image of updated parameters for the merged route: 

<img width="1431" alt="Screenshot 2022-09-15 at 13 22 16" src="https://user-images.githubusercontent.com/86949265/190396678-7ebfbeb3-e08b-4411-94c9-5982da67ddcb.png">


## How Has This Been Tested?

I visually checked that the route was added in the FastAPI interface, and also ran pytests locally. 

- [x] Yes


## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my code and corrected any misspellings

peterdudfield commented on src/gsp.py:
```diff
@@ -58,6 +58,70 @@ async def get_all_available_forecasts(
+# )
+
+
+@router.get("/forecast/{gsp_id}/{only_values}", response_model=Union[Forecast, List[Forecast]])
```
looks good, might be worth moving this `only_values' to a input, but not in the URL. I think its an option but shouldnt always be stated. 

Could rename to only_forecast_values

peterdudfield commented on src/gsp.py:
```diff
@@ -58,6 +58,70 @@ async def get_all_available_forecasts(
+
+    print("this is working")
+
+    only_values_forecast = get_latest_forecast_values_for_a_specific_gsp_from_database(
```
goo to do and 
```
if :

else:
```

Just to make it really clear what is going on

rachel-labri-tipton replied:
i had an else there but then deleted it. i'll put it back in.

peterdudfield replied:
thanks

peterdudfield commented on src/gsp.py:
```diff
@@ -58,6 +58,70 @@ async def get_all_available_forecasts(
+
+        return full_forecast
+
+    print("this is working")
```
Could change this to a logging statement

rachel-labri-tipton commented:
@peterdudfield, all checks have passed! would you be able to have a quick look to see if I missed anything before I merge? 

I'm planning on making a new issue for updating the description for the merged route and could work on that tomorrow or later today.

peterdudfield commented on src/gsp.py:
```diff
@@ -58,12 +58,14 @@ async def get_all_available_forecasts(
+) -> Union[Forecast, List[ForecastValue]]:
     """### Get one forecast for a specific GSP
 
     The return object is a solar forecast with GSP system details.
```
Would you be able to update this documentation. I.e also mention about just get the Forecast Values. Hopefully its just copying stuff from the old documentation

rachel-labri-tipton replied:
I just made a new issue to update the description. probably isn't necessary? I'll go ahead and update the description on this pull request.

peterdudfield replied:
yea, lets keep it together, if thats ok

peterdudfield commented:
Looks, great, just one thing to address. 

Could you also add a screenshot of the routes from the /docs page? You'll have to run it locally and got to the docs page.

rachel-labri-tipton commented:
hey @peterdudfield I updated the route description and added screenshots to the original pull request. would you mind having a quick look at the new description?

peterdudfield commented:
@all-contributors please add @rachel-labri-tipton  for code
